src/output-json-stats.c

```diff
@@ -440,7 +440,14 @@ static OutputInitResult OutputStatsLogInitSub(ConfNode *conf, OutputCtx *parent_
         return result;
     }
 
-    stats_ctx->file_ctx = ajt->file_ctx;
+    SCLogDebug("Preparing file context for stats submodule logger");
+    /* Share output slot with thread 1 */
```

Comment:
why is this done? Can we not just use a thread id for this thread? Asking because this looks like special case handling, and want to avoid that as much as possible

Reply:
The stats submodule is initialized very early in Suricata's startup before any ancillary threads are created.

Stack when this function is invoked:
```
Breakpoint 1, OutputStatsLogInitSub (conf=0x8854783d1efba500, parent_ctx=0x7fffffffdbd0) at output-json-stats.c:387
387	{
(gdb) bt
#0  OutputStatsLogInitSub (conf=0x8854783d1efba500, parent_ctx=0x7fffffffdbd0) at output-json-stats.c:387
#1  0x00005555561bdaf7 in RunModeInitializeEveOutput (conf=0x6060000008c0, parent_ctx=0x60300002e1b0) at runmodes.c:693
#2  0x00005555561bed72 in RunModeInitializeOutputs () at runmodes.c:843
#3  0x000055555587dd25 in PreRunPostPrivsDropInit (runmode=2) at suricata.c:2161
#4  0x0000555555880fba in SuricataMain (argc=11, argv=0x7fffffffe258) at suricata.c:2914
#5  0x000055555587258c in main (argc=11, argv=0x7fffffffe258) at main.c:22
```

Thread(s):
```
(gdb) info threads
  Id   Target Id                                           Frame
* 1    Thread 0x7ffff664df40 (LWP 2053492) "Suricata-Main" OutputStatsLogInitSub (conf=0x8854783d1efba500, parent_ctx=0x7fffffffdbd0)
    at output-json-stats.c:387
```

I can remove the special value `1` and replace it with a value signifying "any" and let `LogFileEnsureExists` use the special value "any" to find a thread slot.

The new call would look like `LogFileEnsureExists(ajt->file_ctx, -1);`

Would you prefer this?